### ISSUE TITLE
ERROR  [nuxt] [request error] [unhandled] [500] The URL must be of scheme file

### ISSUE THREAD

**Issue by ibrahim-990** [nuxt/image#797]
![image](https://user-images.githubusercontent.com/65185082/229298098-8f9f4e40-7728-4e16-a970-09c27405969d.png)

**Comment by ibrahim-990** [comment]
I'm start a new project nuxt3 and add ("@nuxt/image-edge": "^1.0.0-28004510.a0dc8e8").
and on (npm run generate) I get error like in image.

**Comment by ibrahim-990** [comment]
// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
    modules: [
        '@nuxtjs/tailwindcss',
        '@nuxt/image-edge',
    ],
    app: {
        head: {
            title: "Home",
            meta: [
                {name:"description",content:"description website"}
            ],
            link: [
                {
                    rel: "stylesheet",
                    href: 'https://fonts.googleapis.com/css2?family=Almarai:wght@300;400;700;800&display=swap'
                }
            ],
        }
    },
    tailwindcss: {
        // Options
    },
    runtimeConfig: {
        currencyKey: process.env.CURRENCY_API_KEY,
        ipx: {
            dir: "../../public",
        },
    }
})

**Comment by ibrahim-990** [comment]
![image](https://user-images.githubusercontent.com/65185082/229298516-50c058f0-f13d-44d5-a320-06f545ff27a9.png)
This my nuxt.config.ts

**Comment by ibrahim-990** [comment]
![image](https://user-images.githubusercontent.com/65185082/229299765-7997bec8-9481-4cee-a7a4-7dbefb16b628.png)
Here example how to use

**Comment by ibrahim-990** [comment]
@marcelxpfeifer

**Comment by Sadarex** [comment]
Any news on this one?

**Comment by modbender** [comment]
This is happening especially when some kind of SEO package is used, like `nuxt-seo-kit`

I use `nuxt-seo-kit` and I see this error only when it's in my nuxt.config.ts, if I comment out the module it builds successfully.

**Comment by lxccc812** [comment]
How to solve the problem that is still there

**Comment by modbender** [comment]
@harlan-zw Hey this is probably not directly related to you, but
I started getting this error after I used `nuxt-seo-kit`

And to fix it I had to add 

```js
  sitemap: {
    discoverImages: false,
  },
```

**Comment by modbender** [comment]
@harlan-zw Sorry to ping you from an unrelated repository, I have seen your contributions to Nuxt framework, this seems to be happening especially when prerendering and doesn't happen if I comment out

```js
extends: ["nuxt-seo-kit"],
```

Hoping you find a solution in your free time.

**Comment by harlan-zw** [comment]
Thanks for the ping @modbender, happy to try and assist.

The blocker for me is that I'm not on Windows.

If someone can tell me what the relevant code block that the stacktrace points to and add a debug line where it dies (with what is killing it), I should be able to fix it.


```
  at /E:/Project/nuxt/nuxt-nova/.nuxt/prerender/chunks/nitro/nitro-prerenderer.mjs:X
```

(i.e open this file and copy the relevant code at line X)

Before the line calling `fileURLToPath`, put a console.log of what it's trying to convert.

For example this code block

```ts
function readAsset (id) {
  const serverDir = dirname(fileURLToPath(globalThis._importMeta_.url));
  return promises.readFile(resolve(serverDir, assets[id].path))
}
```

Should be this

```ts
function readAsset (id) {
  console.log(globalThis._importMeta_.url)
  const serverDir = dirname(fileURLToPath(globalThis._importMeta_.url));
  return promises.readFile(resolve(serverDir, assets[id].path))
}
```

**Comment by modbender** [comment]
Yeah, I did see that solution somewhere, seems experimental, so switched to WSL

**Comment by JerryChan31** [comment]
I bumped into this problem as well today. Since I don't need provider I just set  `image: { provider: "none" }` in `nuxt.config.ts` and it can passed `nuxt generate` process now. It really takes me some time to find this config option. :-)

**Comment by modbender** [comment]
> I bumped into this problem as well today. Since I don't need provider I just set `image: { provider: "none" }` in `nuxt.config.ts` and it can passed `nuxt generate` process now. It really takes me some time to find this config option. :-)

And with provider none your images are being optimized?

**Comment by JerryChan31** [comment]
> > I bumped into this problem as well today. Since I don't need provider I just set `image: { provider: "none" }` in `nuxt.config.ts` and it can passed `nuxt generate` process now. It really takes me some time to find this config option. :-)
> 
> And with provider none your images are being optimized?

Of course no. But I spent quite some time to fallback since the docs didn’t mention provider none at all, and typescript also didn’t give any hint. So I think leaving a message of fallback option here might help someone who meet this problem.

**Comment by modbender** [comment]
> > > I bumped into this problem as well today. Since I don't need provider I just set `image: { provider: "none" }` in `nuxt.config.ts` and it can passed `nuxt generate` process now. It really takes me some time to find this config option. :-)
> > 
> > 
> > And with provider none your images are being optimized?
> 
> Of course no. But I spent quite some time to fallback since the docs didn’t mention provider none at all, and typescript also didn’t give any hint. So I think leaving a message of fallback option here might help someone who meet this problem.

If the provider is none, no optimization is done, so what's the use of keeping this package installed?

**Comment by JerryChan31** [comment]
> > > > I bumped into this problem as well today. Since I don't need provider I just set `image: { provider: "none" }` in `nuxt.config.ts` and it can passed `nuxt generate` process now. It really takes me some time to find this config option. :-)
> > > 
> > > 
> > > And with provider none your images are being optimized?
> > 
> > 
> > Of course no. But I spent quite some time to fallback since the docs didn’t mention provider none at all, and typescript also didn’t give any hint. So I think leaving a message of fallback option here might help someone who meet this problem.
> 
> If the provider is none, no optimization is done, so what's the use of keeping this package installed?

I'm keeping it for lazy loading, customizable placeholder now. If the nuxt-img is available in SSG mode (in the future) , I can just simply flip the switch on.

**Comment by Voltra** [comment]
Same issues with:

```json
"devDependencies": {
    "@nuxt/content": "^2.8.4",
    "@nuxt/devtools": "latest",
    "@vitejs/plugin-legacy": "^4.1.1",
    "autoprefixer": "^10.4.16",
    "cssnano": "^6.0.1",
    "nuxt": "^3.7.4",
    "nuxt-svgo": "^3.5.4",
    "postcss": "^8.4.30",
    "postcss-easing-gradients": "^3.0.1",
    "postcss-preset-env": "^9.1.4",
    "sass": "^1.68.0",
    "terser": "^5.20.0"
  },
  "dependencies": {
    "@nuxt/image": "1.0.0-rc.2",
    "@studio-freight/lenis": "^1.0.25",
    "foundation-sites": "^6.8.1",
    "gsap": "^3.12.2",
    "lazysizes": "^5.3.2",
    "micromodal": "^0.4.10",
    "node-html-parser": "^6.1.10",
    "striptags": "^3.2.0"
  }
```

**Comment by DamianGlowala** [comment]
If any of you is still affected by this issue in v2, I'll reopen and investigate provided there's a minimal reproduction.
